src/serac/physics/utilities/variational_form/integral.hpp

```diff
@@ -0,0 +1,597 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file integral.hpp
+ *
+ * @brief This file contains the Integral core of the weak form
+ */
+#pragma once
+
+#include "mfem.hpp"
+#include "mfem/linalg/dtensor.hpp"
+
+#include "serac/physics/utilities/variational_form/tensor.hpp"
+#include "serac/physics/utilities/variational_form/quadrature.hpp"
+#include "serac/physics/utilities/variational_form/finite_element.hpp"
+#include "serac/physics/utilities/variational_form/tuple_arithmetic.hpp"
+
+namespace serac {
+
+namespace impl {
```

Comment:
Minor: Any thoughts on `impl` vs `detail`?  I think we definitely have `detail` elsewhere, not sure how important consistency is.  I think `impl` is clearer/more accurate but in my experience `detail` is more common in C++ codebases

Reply:
I have no preference on `impl` versus `detail`, and I agree that it would be better to be consistent!

Reply:
Yeah, we use `detail` elsewhere and I couldn't find `impl`, so we should use `detail` here.